### PR TITLE
Replace pow(, 2) with square

### PR DESCRIPTION
**Pull request by naoyam**
Just a small, localized optimization. Originally mentioned by a potential user trying the fuser with GNN models.

### CODE REVIEW

**Review by rdspring1 — COMMENTED** [review]
LGTM. 

Transforming `pow(x, 3)` with `x * x * x` was also necessary for tanh gelu performance. 
Would it be possible to add that to this PR?

**Comment by naoyam** [comment]
> Transforming pow(x, 3) with x * x * x was also useful for tanh gelu performance.
Would it be possible to add that to this PR?

Hmm, it would be no longer a single binary op. If we are using `BinaryOpType::Mul`, we would need an intermediate tensor, but automatically introducing an additional tensor at the time of lowering would need to be done more carefully. 

It would be trivial to do if this were done at the codegen time, i.e., within `CudaKernelGenerator`, although I'd generally prefer doing replacements within the IR. Do you need a kernel IR where `pow` is replaced? Or just replaced CUDA code is enough?

**Comment by rdspring1** [comment]
Replacing CUDA code is good enough. Maybe we could mark the pow function if it is transformed.

**Comment by naoyam** [comment]
Updated. `pow(, 3)` is also replaced. Since the replacement is done when generating a kernel string, no verification of replacement is done.

File: torch/csrc/jit/codegen/cuda/codegen.cpp [hunk 1]
```diff
@@ -518,7 +519,69 @@ class CudaKernelGenerator : private kir::IrVisitor {
...
+
+    if (print_inline_) {
+      code_ << lhs << " * " << lhs;
+      if (exponent.value() == 3) {
+        code_ << " * " << lhs;
+      }
+    } else {
+      indent() << gen(node->out());
+      if (node->out()->isScalar()) {
+        code_ << " = " << lhs << " * " << lhs;
```
Comment (rdspring1) [inline torch/csrc/jit/codegen/cuda/codegen.cpp:563]: Is there a `exponent.value() == 3` case here?

**Review by rdspring1 — APPROVED** [review]
LGTM

File: torch/csrc/jit/codegen/cuda/codegen.cpp (hunk 1 above)
Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/codegen.cpp:563]: Oops, thanks. Fixed. I also extended the test to exercise this code path.

**Pull request merged**